### PR TITLE
Sw dspdc 709 File metadata extraction for ENCODE

### PR DESCRIPTION
**Pull request by snwessel**
Extracting file metadata and updating comments ([Jira Ticket](https://broadinstitute.atlassian.net/browse/DSPDC-709))

### CODE REVIEW

File: src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala [hunk 1]
```diff
@@ -42,10 +42,10 @@ object ExtractionPipeline {
...
       *
-      * @param encodeEntity the entity from which to extract a field
-      * @param referenceField the field name to use as an identifier in getIds
-      * @param prevData the data to base the ID extraction off of
-      * @param entityFieldName the field name to extract if it isn't @id
+      * @param encodeEntity the entity which should be extracted
+      * @param referenceField the field name to use as an identifier for the encodeEntity (in getIds)
+      * @param prevData the parent entity (which references a set of encodeEntities) from which ID extraction will be based
+      * @param entityFieldName the encodeEntity field name which references the parent (default is @id)
       * @return the extracted linked entities
```
Comment (snwessel) [inline src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala]: @raaidbroad let me know if you have any other thoughts on how these comments could be phrased better

File: src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala [hunk 2]
```diff
@@ -99,14 +99,20 @@ object ExtractionPipeline {
...
       "experiment",
       replicates
     )
 
+    // don't need to use files apart from storing them, so we don't assign an output here
+    extractLinkedEntities(
+      EncodeEntity.File,
+      "@id",
+      experiments,
+      "dataset"
```
Comment (danxmoran) [inline src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala]: Existing code, but: It's tough to know what these different `String` args correspond to when just looking at the call-sites. Can you update the call(s) to use a kwargs-style call?
```scala
extractLinkedEntities(
  encodeEntity = EncodeEntity.File,
  referenceField = "@id",
  prevData = experiments,
  entityFieldName = "dataset"
)
```

File: src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala [hunk 3]
```diff
@@ -42,10 +42,10 @@ object ExtractionPipeline {
...
       * Generic helper method for extracting linked entities and saving them.
       *
-      * @param encodeEntity the entity from which to extract a field
-      * @param referenceField the field name to use as an identifier in getIds
-      * @param prevData the data to base the ID extraction off of
-      * @param entityFieldName the field name to extract if it isn't @id
+      * @param encodeEntity the entity which should be extracted
+      * @param referenceField the field name to use as an identifier for the encodeEntity (in getIds)
+      * @param prevData the parent entity (which references a set of encodeEntities) from which ID extraction will be based
+      * @param entityFieldName the encodeEntity field name which references the parent (default is @id)
```
Comment (danxmoran) [inline src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala]: Existing code, but: I think these argument names could be much more clear. Can we rename them? I'm thinking something like:
* `encodeEntity` -> `entityToExtract`
* `referenceField` -> `matchingField`
* `prevData` -> `linkingEntities`
* `entityFieldName` -> `linkField`

Comment (danxmoran) [reply src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala]: (Also, to be clear: naming debates are open to all. Please counter-suggest if you don't like one of mine 😄)

File: src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala (hunk 2 above)
Comment (danxmoran) [reply src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala]: (The kwargs would need to be updated to match the new arg names, if they get changed)

File: src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala (hunk 3 above)
Comment (snwessel) [reply src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala]: I like your suggestions 😁... I had been struggling to come up with something to replace prevData

Comment (snwessel) [reply src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala]: Although I might change `linkField` to `linkedField` unless you object

File: src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala [hunk 4]
```diff
@@ -42,32 +42,35 @@ object ExtractionPipeline {
...
-      * @param encodeEntity the entity from which to extract a field
-      * @param referenceField the field name to use as an identifier in getIds
-      * @param prevData the data to base the ID extraction off of
-      * @param entityFieldName the field name to extract if it isn't @id
+      * @param entityToExtract the entity which should be extracted
+      * @param matchingField name of the field to use as an identifier for the entityToExtract
+      *                      (should match the id that's used in the linkedField)
+      * @param linkingEntities the parent entity (which references a set of encodeEntities)
+      *                        from which ID extraction will be based
+      * @param linkedField name of the entityToExtract field which references the linkingEntities (default is @id)
```
Comment (raaidbroad) [inline src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala]: 🙌 much better var names and comments ty for updating

**Pull request merged**